producer/event_producer.py

```diff
@@ -1,29 +1,43 @@
 import json
+import base64
+
 from abc import ABC
 from typing import Type, Union
 
+import cachetools
+
 import psycopg2
+from psycopg2 import pool
 from psycopg2.extras import LogicalReplicationConnection
 
 from common.event import BaseEvent
 from common.qconnector import QConnector
 
 from common.log import get_logger
 
+from common.utils import DeserializerUtils as parser_utils
+
+
 logger = get_logger(__name__)
 
+# 12 hour cache
+# Maximum of 1024 entries
+table_name_cache = cachetools.TTLCache(maxsize=1024, ttl=43200)
```

Comment:
can move ttl to env variable, easier to change later on
the 12 hour of tablename looks too high